src/tests/translation/number_binop.rs

```diff
@@ -0,0 +1,701 @@
+        assert_eq!(program.values.len(), 9);
+
+        assert_eq!(unwrap_fragment!(program.values[0].clone(), Raw).value, "echo ");
+        let left_subprocess_outer = unwrap_fragment!(program.values[1].clone(), Subprocess);
+        let left_program_outer = unwrap_fragment!(*left_subprocess_outer.fragment.clone(), List);
+        assert_eq!(left_program_outer.values.len(), 9);
+        assert_eq!(unwrap_fragment!(left_program_outer.values[0].clone(), Raw).value, "echo ");
+
+        let left_subprocess_inner = unwrap_fragment!(left_program_outer.values[1].clone(), Subprocess);
+        let left_program_inner = unwrap_fragment!(*left_subprocess_inner.fragment.clone(), List);
+        assert_eq!(left_program_inner.values.len(), 9);
+        assert_eq!(unwrap_fragment!(left_program_inner.values[0].clone(), Raw).value, "echo ");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[1].clone(), Raw).value, "100.0");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[2].clone(), Raw).value, " '%' ");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[3].clone(), Raw).value, "7.5");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[4].clone(), Raw).value, " | bc ");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[5].clone(), Raw).value, "");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[6].clone(), Raw).value, " | sed '");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[7].clone(), Raw).value, "/\\./ s/\\.\\{0,1\\}0\\{1,\\}$//");
+        assert_eq!(unwrap_fragment!(left_program_inner.values[8].clone(), Raw).value, "'");
+
+        assert_eq!(unwrap_fragment!(left_program_outer.values[2].clone(), Raw).value, " '%' ");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[3].clone(), Raw).value, "3.0");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[4].clone(), Raw).value, " | bc ");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[5].clone(), Raw).value, "");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[6].clone(), Raw).value, " | sed '");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[7].clone(), Raw).value, "/\\./ s/\\.\\{0,1\\}0\\{1,\\}$//");
+        assert_eq!(unwrap_fragment!(left_program_outer.values[8].clone(), Raw).value, "'");
+
+        assert_eq!(unwrap_fragment!(program.values[2].clone(), Raw).value, " '%' ");
+        assert_eq!(unwrap_fragment!(program.values[3].clone(), Raw).value, "2.5");
+        assert_eq!(unwrap_fragment!(program.values[4].clone(), Raw).value, " | bc ");
+        assert_eq!(unwrap_fragment!(program.values[5].clone(), Raw).value, "");
+        assert_eq!(unwrap_fragment!(program.values[6].clone(), Raw).value, " | sed '");
+        assert_eq!(unwrap_fragment!(program.values[7].clone(), Raw).value, "/\\./ s/\\.\\{0,1\\}0\\{1,\\}$//");
+        assert_eq!(unwrap_fragment!(program.values[8].clone(), Raw).value, "'");
+    }
+}
```

Comment:
This seems like a lot of work, to write all of this. Why not compare snapshots? It will be easier to read, and will showcase everything. No change will go unnoticed